### PR TITLE
[Typing][C-84] Add type annotations for `python/paddle/incubate/nn/layer/fused_linear.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

[为公开 API 标注类型提示信息](https://github.com/PaddlePaddle/Paddle/issues/65008)
C-84 python/paddle/incubate/nn/layer/fused_linear.py

### CODE REVIEW

File: python/paddle/incubate/nn/layer/fused_linear.py [hunk 1]
```diff
@@ -67,15 +74,19 @@ class FusedLinear(Layer):
...
-        bias_attr=None,
-        transpose_weight=False,
-        name=None,
+        in_features: int,
+        out_features: int,
+        weight_attr: ParamAttrLike | None = None,
+        bias_attr: ParamAttrLike | None = None,
+        transpose_weight: bool = False,
+        name: str | None = None,
     ):
```
Comment (megemini) [inline python/paddle/incubate/nn/layer/fused_linear.py]:
```suggestion
    ) -> None:
```

**Review by megemini — COMMENTED** [review]
CI 示例代码运行错误的问题我这儿木有 ～ 合一下代码再试试？

File: python/paddle/incubate/nn/layer/fused_linear.py (hunk 1 above)
Comment (enkilee) [reply python/paddle/incubate/nn/layer/fused_linear.py]: 收到

**Review by megemini — APPROVED** [review]
LGTM ～ 🤟

**Comment by SigureMo** [comment]
codestyle 没过，需要重新 format 下

**Review by luotao1 — APPROVED** [review]

**Pull request merged**